astrolabe/utils.py

```diff
@@ -300,31 +281,67 @@ def get_logs(admin_client, project, cluster_name):
         logTypes=['FTDC','MONGODB'],#,'AUTOMATION_AGENT','MONITORING_AGENT','BACKUP_AGENT'],
         sizeRequestedPerFileBytes=100000000,
     )
-    data = admin_client.groups[project.id].logCollectionJobs.post(**params).data
-    job_id = data['id']
+
+    # Wait 10 minutes for logs to get collected.
+    # This is a guess as to how long job collection might take, we haven't
+    # investigated the actual times over a sufficiently large sample size.
+    # The same timeout is used for retrying collection jobs and waiting for
+    # any single collection job - if this turns out to be problematic,
+    # we'll need to do some analysis of actual times to refine the timeouts.
+    timeout = 600
+    local = {}
     
-    while True:
-        LOGGER.debug('Poll job %s' % job_id)
-        data = admin_client.groups[project.id].logCollectionJobs[job_id].get().data
-        if data['status'] == 'IN_PROGRESS':
-            sleep(1)
-        elif data['status'] == 'SUCCESS':
-            break
-        else:
-            raise Exception("Unexpected log collection job status %s" % data['status'])
+    def collect():
+        try:
+            data = admin_client.groups[project.id].logCollectionJobs.post(**params).data
+            job_id = data['id']
+            
+            def check():
+                data = admin_client.groups[project.id].logCollectionJobs[job_id].get().data
+                if data['status'] == 'SUCCESS':
+                    local['data'] = data
+                    return True
+                elif data['status'] != 'IN_PROGRESS':
+                    raise AstrolabeTestCaseError("Unexpected log collection job status: %s: %s" % (data['status'], data))
```

Comment:
Can you add an `else` block here? The method returns `None` when the status is `IN_PROGRESS` which allows the `poller()` method to keep polling but ideally, we should be returning a bool `False` from `check()` explicitly not implicitly.
```
elif data['status'] != 'IN_PROGRESS':
    raise AstrolabeTestCaseError("Unexpected log collection job status: %s: %s" % (data['status'], data))
else:
    # status == 'IN_PROGRESS', continue polling for logs to be ready
    return False
```